### ISSUE TITLE
Building hevea without ocamlbuild

### ISSUE THREAD

**Issue by cessenat** [maranget/hevea#59]
Hello,
In Hevea "README" it is indicated that one can choose TARGET=byte if "ocamlbuild" is not installed.
But, in that case, script "ocn.sh" still calls "ocamlbuild".
Is there a way to correct that ?
Thanks.
Olivier Cessenat.

**Comment by maranget** [comment]
Hello,
The README file looks completely outdated, as it does not even mention ocamlbuild. This file surely needs an update.

However ocamlbuild is required for compiling hevea, ocamlbuild is easily installed with `opam`.

**Comment by cessenat** [comment]
Thanks for answering that fast. As you mention the README, the Makefile 
and install.sh also keep traces of "byte" TARGET rule, which lead me to 
think it could still be valid.

So, we actually need to assume hevea requires ocamlbuild which itself 
requires ocaml, right ?

Le 08/02/2021 à 20:11, Luc Maranget a écrit :

**Comment by maranget** [comment]
> So, we actually need to assume hevea requires ocamlbuild which itself requires ocaml, right ?

You are correct, I have slightly corrected the README file, so as to state that ocaml and ocamlbuild are required for building hevea. That said, hevea is on `opam` (the OCAML  package manager) and the opam version is up to date. Hence, having installed opam, installing hevea is as simple as
```
opam install hevea
```

**Comment by cessenat** [comment]
Thanks. My aim is to make an spack package description for hevea. I've 
already just made a PR for ocamlbuild. Ocaml itself is already in spack.

Maybe you would prefer to add hevea yourself, in what case I can provide 
my file to you, what do you think ?

PS: I use hevea for ftagshtml for instance.

Le 09/02/2021 à 11:48, Luc Maranget a écrit :

**Comment by maranget** [comment]
Fine, I'd rather have you to take care of hevea spack package if you don't mind.... Do not hesitate to fill an issue if some problem still occurs.

Thanks

**Comment by cessenat** [comment]
OK I'll do it ; for the moment waiting for my PR to be accepted for 
ocamlbuild, while learning to use git and github...

Le 10/02/2021 à 17:50, Luc Maranget a écrit :

**Comment by cessenat** [comment]
By the way, do you consider switching to "Dune" in the future, as the ocambuild pages seems to advise it ?

**Comment by cspiel** [comment]
The build process already relies on GNU(?) Make e.g. to launch ocamlbuild(1)
or to build the documentation.  The project's directory structure is so simple
that we could remove the ocamlbuild dependency and solely construct
Hevea with Make.  The heavy lifting, i.e. computing the module dependencies
can done by ocamldep(1), which is part of the OCaml compiler suite.

**Comment by maranget** [comment]
Hi @cspiel, `ocamlbuild` is more robust to changes than plain `Makefile`.  It is also simpler. However transition to `dune`, a more modern and maintained tool is an option.
